src/store/modules/SplashModule.ts

```diff
@@ -59,21 +84,23 @@ export const SplashModule = {
                 dispatch('tsMods/syncPackageList', null, {root: true});
             }
         },
-        async fetchPackageListIndex({state, dispatch}): Promise<PackageListIndex | undefined> {
-             state.splashText = 'Checking for mod list updates from Thunderstore';
+        async fetchPackageListIndex({commit, dispatch}): Promise<PackageListIndex | undefined> {
+             commit('setSplashText', 'Checking for mod list updates from Thunderstore');
 
             try {
                 return await dispatch('tsMods/fetchPackageListIndex', null, {root: true});
             } catch (e) {
                 console.error('SplashModule failed to fetch mod list index from API.', e);
                 return undefined;
             } finally {
-                const requestItem = await dispatch('getRequestItem', 'PackageListIndex') as RequestItem;
-                requestItem.setProgress(100);
+                await dispatch('updateRequestItem', {
+                    requestName: 'PackageListIndex',
+                    value: 100
+                } as UpdateRequestItemBody);
```

Comment:
Is this an attempt to try to improve the type safety e.g. in case of using `value: "100"` or what?

Reply:
No, we aren't allowed to modify the RequestItem object in the same way now because VueX has added constraints in the later versions. We have to do these changes in a commit (this dispatch goes to a commit, related to the comment above - this one could commit directly though).

The actual type definitions are just because we don't have a use for anything other than number, and so make sense to have anyway.